graph/adjacency/ThingAdjacency.java

```diff
@@ -37,18 +45,32 @@
      */
     ThingIteratorBuilder edge(Encoding.Edge.Thing encoding);
 
+
     /**
      * Returns an {@code IteratorBuilder} to retrieve vertices of a set of edges.
      *
      * This method allows us to traverse the graph, by going from one vertex to
      * another, that are connected by edges that match the provided {@code encoding}
-     * and {@code lookahead}.
+     * and {@code lookAhead}.
      *
      * @param encoding  type of the edge to filter by
      * @param lookAhead information of the adjacent edge to filter the edges with
      * @return an {@code IteratorBuilder} to retrieve vertices of a set of edges.
      */
-    ThingIteratorBuilder edge(Encoding.Edge.Thing encoding, IID... lookAhead);
+    default ThingIteratorSortedBuilder edge(Encoding.Edge.Thing encoding, IID... lookAhead) {
+        if (encoding == Encoding.Edge.Thing.HAS) return edgeHas(lookAhead);
+        else if (encoding == Encoding.Edge.Thing.PLAYING) return edgeHas(lookAhead);
+        else if (encoding == Encoding.Edge.Thing.RELATING) return edgeHas(lookAhead);
+        else if (encoding == Encoding.Edge.Thing.ROLEPLAYER) {
+            if (lookAhead.length > 0) return edgeRolePlayer(lookAhead[0], Arrays.copyOfRange(lookAhead, 1, lookAhead.length));
+            else throw TypeDBException.of(ILLEGAL_OPERATION);
+        }
+        else throw TypeDBException.of(ILLEGAL_STATE);
+    }
+    ThingIteratorSortedBuilder edgeHas(IID... lookAhead);
+    ThingIteratorSortedBuilder edgePlaying(IID... lookAhead);
+    ThingIteratorSortedBuilder edgeRelating(IID... lookAhead);
+    ThingIteratorSortedBuilder edgeRolePlayer(IID roleType, IID... lookAhead);
```

Comment:
only under certain circumstances can we provide a sorted edge iterator - specifically any `has`/`relating`/`playing` edge iterator, or a `roleplayer` edge when also given a `roleType` lookahead. We enforce this via the function signature